dartagnan/src/main/java/com/dat3m/dartagnan/expression/type/AggregateType.java

```diff
@@ -2,33 +2,58 @@
 
 import com.dat3m.dartagnan.expression.Type;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
-public final class AggregateType implements Type {
+import static com.dat3m.dartagnan.expression.type.TypeFactory.paddedSize;
 
-    private final List<Type> fields;
+public class AggregateType implements Type {
 
-    AggregateType(List<Type> directFields) {
-        this.fields = List.copyOf(directFields);
+    private static final TypeFactory types = TypeFactory.getInstance();
+
+    private final List<TypeOffset> directFields;
+
+    AggregateType(List<Type> fields) {
+        this(fields, computeDefaultOffsets(fields));
+    }
```

Comment:
I think its nicer to get rid of this constructor and instead have a corresponding overload in `TypeFactory.makeAggregateType` (the used helper functions would also move to `TypeFactory` of course). This should also remove the reference to `TypeFactory` in this class altogether.